pkg/gofr/crud_handlers.go

```diff
@@ -61,7 +61,16 @@ type entity struct {
 
 // scanEntity extracts entity information for CRUD operations.
 func scanEntity(object interface{}) (*entity, error) {
-	entityType := reflect.TypeOf(object).Elem()
+	if object == nil {
+		return nil, errInvalidObject
```

Comment:
Can we find some way to include the struct details ,
So thinking of a scenario where user has used method `AddRESTHandlers` for 2 different structs, one he has passed by value and other by pointer.

So when we will run the code, w err message : ` invalid object for AddRESTHandlers` will not give clear description of which object is invalid....